### PR TITLE
Update help window

### PR DESCRIPTION
**Pull request by KSSWSept20**
- Add a command summary table to the help window.
- Enable user to close the window easily by typing 'q'.

### CODE REVIEW

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 1]
```diff
@@ -99,4 +133,44 @@ private void copyUrl() {
         url.putString(USERGUIDE_URL);
         clipboard.setContent(url);
     }
+
+    /**
+     * Fills the command summary table with all available commands.
+     */
+    private void fillCommandSummaryTable() {
+        guideItems.add(new UserGuideItem("help", "Call this user guide.\n"
+                + "Can also call by typing F1 on keyboard"));
```
Comment (Tanzhiheng26) [inline src/main/java/seedu/address/ui/HelpWindow.java:142]:
```suggestion
        guideItems.add(new UserGuideItem("help", "Open command summary (can also be accessed by pressing F1 key)"));
```

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 2]
```diff
@@ -16,7 +24,11 @@
 public class HelpWindow extends UiPart<Stage> {
 
     public static final String USERGUIDE_URL = "https://se-education.org/addressbook-level3/UserGuide.html";
-    public static final String HELP_MESSAGE = "Refer to the user guide: " + USERGUIDE_URL;
+    public static final String HELP_MESSAGE = "For more detail, refer to the user guide:\n" + USERGUIDE_URL
+            + "\n\nThings to note on:"
+            + "\n  • Items in square brackets are optional."
+            + "\n  • Items with '...' after them can be used multiple times including zero times."
+            + "\n  • Close this window by typing 'q' on keyboard.";
```
Comment (Tanzhiheng26) [inline src/main/java/seedu/address/ui/HelpWindow.java:31]:
```suggestion
            + "\n  • Press 'q' to close this window.";
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -83,6 +83,7 @@ CulinaryContacts is a **desktop app for managing contacts, optimized for use via
 ### Viewing help : `help`
 
 Shows the full command summary of CulinaryContacts at a glance. 
+Closes the help window by typing 'q' on keyboard.
```
Comment (Tanzhiheng26) [inline docs/UserGuide.md:86]:
```suggestion
Press 'q' to close the help window.
```

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 4]
```diff
@@ -99,4 +133,44 @@ private void copyUrl() {
...
+        guideItems.add(new UserGuideItem("list", "List all contacts.\n"));
+        guideItems.add(new UserGuideItem("edit", "Edit a contact.\n"
+                + "edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…\n"
+                + "e.g., edit 1 p/91234567 e/johndoe@example.com"));
+        guideItems.add(new UserGuideItem("find", "Find contacts with a keyword.\n"
+                + "find KEYWORD [MORE_KEYWORDS]\n"
+                + "e.g., find John"));
+        guideItems.add(new UserGuideItem("filter", "Filter contacts with specified tags.\n"
+                + "filter TAG [MORE_TAGS]\n"
+                + "e.g., filter supplier"));
```
Comment (Tanzhiheng26) [inline src/main/java/seedu/address/ui/HelpWindow.java:156]:
```suggestion
                + "e.g., filter supplier seafood"));
```

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 5]
```diff
@@ -99,4 +133,44 @@ private void copyUrl() {
...
+                + "Can also call by typing F1 on keyboard"));
+        guideItems.add(new UserGuideItem("add", "Add a new contact.\n"
+                + "add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]…\n"
+                + "e.g., add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/supplier "
+                + "t/seafood"));
+        guideItems.add(new UserGuideItem("list", "List all contacts.\n"));
+        guideItems.add(new UserGuideItem("edit", "Edit a contact.\n"
+                + "edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…\n"
+                + "e.g., edit 1 p/91234567 e/johndoe@example.com"));
+        guideItems.add(new UserGuideItem("find", "Find contacts with a keyword.\n"
```
Comment (Tanzhiheng26) [inline src/main/java/seedu/address/ui/HelpWindow.java:151]:
```suggestion
        guideItems.add(new UserGuideItem("find", "Find contacts whose names contain any of the given keywords.\n"
```

**Review by Tanzhiheng26 — CHANGES_REQUESTED** [review]
Help window looks good, just a few nits to fix.

**Review by Tanzhiheng26 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 6]
```diff
@@ -15,8 +23,12 @@
  */
 public class HelpWindow extends UiPart<Stage> {
 
-    public static final String USERGUIDE_URL = "https://se-education.org/addressbook-level3/UserGuide.html";
-    public static final String HELP_MESSAGE = "Refer to the user guide: " + USERGUIDE_URL;
+    public static final String USERGUIDE_URL = "https://ay2324s2-cs2103t-w09-3.github.io/tp/UserGuide.html";
+    public static final String HELP_MESSAGE = "For more details, refer to the user guide:\n" + USERGUIDE_URL
+            + "\n\nThings to note:"
```
[the remark this answers is not in the document]
Comment (ziiqii) [reply src/main/java/seedu/address/ui/HelpWindow.java:28]: Could be useful to inform users of the sort function in the help window

**Review by ziiqii — COMMENTED** [review]
Overall help window looks good, I've left some comments on areas that need fixing.

Some comments for your consideration (for future iterations):
1. Could potentially remove sorting from the 'Usage' header in the help window, as users don't usually sort using that field.
2. The default order of commands shown initially (help -> add -> list...) becomes inaccessible once the user sorts the commands, perhaps we could restore it, as the default order corresponds to the order of features in the UG.
3. Could consider styling parts of the command usage with markdown, similar to our UG, improving readability.

**Review by ziiqii — APPROVED** [review]
LGTM!

**Pull request merged**